Intersect.Server/Entities/Events/CommandProcessing.cs

```diff
@@ -707,10 +707,21 @@ Stack<CommandInstance> callStack
             Stack<CommandInstance> callStack
         )
         {
-            player.Warp(
-                command.MapId, command.X, command.Y,
-                command.Direction == WarpDirection.Retain ? (byte) player.Dir : (byte) (command.Direction - 1)
-            );
+            if (command.ChangeInstance)
+            {
+                player.Warp(
+                    command.MapId, command.X, command.Y,
+                    command.Direction == WarpDirection.Retain ? (byte)player.Dir : (byte)(command.Direction - 1),
+                    mapInstanceType: command.InstanceType
+                );
+            }
+            else
+            {
+                player.Warp(
+                    command.MapId, command.X, command.Y,
+                    command.Direction == WarpDirection.Retain ? (byte)player.Dir : (byte)(command.Direction - 1)
+                );
+            }
```

Comment:
Per my last comment all of this code doesn't need to be repeated if you make the `mapInstanceType: ` a ternary (`? : `)

Reply:
Doesn't appear I can do those in our version of C#, as function params.